src/server/test/routes/authenticatorParamsTest.js

```diff
@@ -0,0 +1,215 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+const { expect } = require('chai');
+const { chai, mocha, app, testDB } = require('../common');
+const { validateString, validateToken, testInvalidField } = require('../util/validationHelpers');
+
+// Note: authenticator.js primarily contains middleware functions, not direct API endpoints
+// However, the credentialsRequestValidationMiddleware is used by other routes that accept username/password
+// We'll test the validation logic through a mock endpoint or by testing routes that use it
+
+mocha.describe('Authenticator Parameter Validation', () => {
+
+    // Test the credentials validation used by login and obvius endpoints
+    mocha.describe('Credentials Validation (username/password)', () => {
+        // Since authenticator.js doesn't export direct endpoints, we test through routes that use it
+        // The login route uses credentialsRequestValidationMiddleware
+        const LOGIN_ENDPOINT = '/api/login';
+        
+        const baseCredentials = {
+            username: 'validuser',
+            password: 'validpass123'
+        };
+
+        mocha.it('should validate username field', async () => {
+            await validateString({
+                field: 'username',
+                endpoint: LOGIN_ENDPOINT,
+                basePayload: baseCredentials,
+                required: true,
+                minLength: 5,
+                maxLength: 254
+            });
+        });
+
+        mocha.it('should validate password field', async () => {
+            await validateString({
+                field: 'password',
+                endpoint: LOGIN_ENDPOINT,
+                basePayload: baseCredentials,
+                required: true,
+                minLength: 8,
+                maxLength: 1000 // Security: prevent DoS via huge passwords
+            });
+        });
+
+        mocha.it('should reject payloads with extra fields (parameter injection)', async () => {
+            const payloadWithExtra = {
+                ...baseCredentials,
+                maliciousField: 'injection attempt',
+                anotherField: 'should be rejected'
+            };
+
+            const res = await chai.request(app)
+                .post(LOGIN_ENDPOINT)
+                .send(payloadWithExtra);
+            expect(res).to.have.status(400);
+        });
+
+        mocha.it('should handle malicious username inputs', async () => {
```

Comment:
I like the idea of these tests. These seem to test invalid fields which are tested elsewhere. I'm wondering if the test should try to pass a command in the username, password, etc. to see if OED might execute it.